### ISSUE TITLE
cy.intercept not working in Angular component test, works in e2e test

### ISSUE THREAD

**Issue by bahmutov** [cypress-io/cypress#23479]
### Current behavior

I set up an angular component test with `cy.intercept`, and it does not appear to do anything

```js
import { HttpClientModule } from '@angular/common/http';
import { TwainComponent } from './../twain/twain.component';
import { TwainService } from '../twain/twain.service';
import {AboutComponent} from './about.component'

describe('About', () => {
  it('mounts', () => {
    cy.intercept({
      method: 'GET',
      pathname: '/api/quotes/1'
    }, {
      statusCode: 200,
      body: {
        id: 1,
        quote: 'This is a test quote'
      }
    })
    cy.mount(AboutComponent, {
      imports: [HttpClientModule],
      declarations: [TwainComponent],
      providers: [TwainService]
    })
  })
})
```

<img width="938" alt="Screen Shot 2022-08-20 at 16 38 24" src="https://user-images.githubusercontent.com/2212006/185765231-1966f2c7-2a64-45a3-9780-b9f7f8e1c21d.png">

The same intercept works fine in E2E test

```js
it('loads the quote', () => {
  cy.intercept({
    method: 'GET',
    pathname: '/api/quotes/1'
  }, {
    body: {
      id: 1,
      quote: 'This is a test quote'
    }
  }).as('quote')
  cy.visit('/about')
  cy.wait('@quote')
  cy.contains('This is a test quote').should('be.visible')
})
```
<img width="1070" alt="Screen Shot 2022-08-20 at 16 40 30" src="https://user-images.githubusercontent.com/2212006/185765294-dbeb3fdb-f150-4257-b0ff-f0013be77884.png">

### Desired behavior

`cy.intercept` should work in Angular component tests

### Test code to reproduce

Repo and commit https://github.com/bahmutov/test-tour-of-heroes/commit/443d21a

- clone the repo
- check out the commit 443d21a
- `npm install`
- `npx cypress open`
- run the about component test
The test fails because the network call is not intercepted

### Cypress Version

10.6.0

### Node version

v16

### Operating System

mac

### Debug Logs

```shell
Using `DEBUG=cypress:net-stubbing*`

When using E2E test

  cypress:net-stubbing:server:driver-events received driver event { eventName: 'route:added', args: [ 'route:added', { routeId: '1661027486779-12', hasInterceptor: false, routeMatcher: [Object], staticResponse: [Object] } ] } +0ms
  cypress:net-stubbing:server:intercept-response InterceptResponse { req: { url: '/about' }, request: undefined } +2s
GET /about 200 19.288 ms - -
  cypress:net-stubbing:server:intercept-response InterceptResponse { req: { url: '/styles.css' }, request: undefined } +47ms
GET /styles.css 200 21.985 ms - -
  cypress:net-stubbing:server:intercept-response InterceptResponse { req: { url: '/styles.js' }, request: undefined } +10ms
  cypress:net-stubbing:server:intercept-response InterceptResponse { req: { url: '/runtime.js' }, request: undefined } +24ms
  cypress:net-stubbing:server:intercept-response InterceptResponse { req: { url: '/polyfills.js' }, request: undefined } +2ms
  cypress:net-stubbing:server:intercept-response InterceptResponse { req: { url: '/vendor.js' }, request: undefined } +2ms
  cypress:net-stubbing:server:intercept-response InterceptResponse { req: { url: '/main.js' }, request: undefined } +2ms
GET /runtime.js 200 58.683 ms - -
GET /main.js 200 60.874 ms - -
GET /styles.js 200 18.875 ms - -
GET /polyfills.js 200 58.101 ms - -
GET /vendor.js 200 60.049 ms - -
  cypress:net-stubbing:server:intercept-request intercepting request { requestId: 'interceptedRequest14', req: { url: '/api/quotes/1' } } +0ms
  cypress:net-stubbing:server:util sending event to driver { eventName: 'before:request', data: { eventId: 'event15', subscription: { eventName: 'before:request', await: false, routeId: '1661027486779-12' }, browserRequestId: '40572.97', requestId: 'interceptedRequest14', data: { headers: [Object], url: 'http://localhost:4200/api/quotes/1', method: 'GET', httpVersion: '1.1', body: '' } } } +0ms
```

When using component test, terminal shows very few messages

```
GET /__cypress/src/cypress_support_component_ts.js 200 12.624 ms - -
GET /__cypress/src/spec-0.js 200 5.724 ms - -
  cypress:net-stubbing:server:driver-events received driver event { eventName: 'route:added', args: [ 'route:added', { routeId: '1661027507779-6', hasInterceptor: false, routeMatcher: [Object], staticResponse: [Object] } ] } +21s
GET /__cypress/iframes//Users/glebbahmutov/git/test-tour-of-heroes/src/app/about/api/quotes/1 200 2.566 ms - 605
```

Nothing else
```
```

### Other

Angular v14

**Comment by jordanpowell88** [comment]
Hey @bahmutov. `cy.intercept` is working in Angular. You can see our ([demo app example](https://github.com/cypress-io/cypress-component-testing-apps/blob/main/angular/src/app/app.component.cy.ts#L18)).

I was able to pull down your repo and identify what is going on. Currently `TwainService.getQuote()` makes an http request using `HttpClient` to a relative path of `api/quotes/1` but Angular is treating this like a route change which is why it's returning a 304....see image below

![Screen Shot 2022-08-23 at 2 17 54 PM](https://user-images.githubusercontent.com/3605268/186251822-db004d25-7ed8-4838-b227-d2ecc05e8c46.png)

If you change the api request url to be something like the following: `http://mydumbyapiendpoint.com/api/quotes/1` and then change your `intercept` url to `**/api/quotes/1` it should be behaving as expected

**Comment by bahmutov** [comment]
Oh my goodness, thank you @jordanpowell88 of course, `http.get` is a route change, makes sense /s

**Comment by blessanm86** [comment]
Hi @jordanpowell88 , I just faced the same issue but I'm using react+react-query+axios.
Ur fix worked, I added a prefix like `https://dummy.com`.

But I'm not sure why the `304` and why intercept doesn't work without a prefix?

I actually didnt understand this line `but Angular is treating this like a route change which is why it's returning a 304.`. I assume its the same issue in react. I just dont understand how an ajax call is seen as a route change?
Can u explain a bit? Also, I think it would be great if this was mentioned in the docs.
